From a13a83465c685d6ec8d47b6f10646986ded16a68 Mon Sep 17 00:00:00 2001
From: Mark Reynolds <mreynolds@redhat.com>
Date: Wed, 9 May 2018 16:36:48 -0400
Subject: [PATCH] Ticket 49665 - Upgrade script doesn't enable PBKDF2 password
 storage plug-in

Description:  There is no upgrade script to add the PBKDF2 plugin, this
              fix adds the script.

https://pagure.io/389-ds-base/issue/49665

Reviewed by: ?

(cherry picked from commit dc690dd231a626b3b6a2019fee51e3cb15db7962)
---
 Makefile.am                                          |  1 +
 ldap/admin/src/scripts/50pbkdf2pwdstorageplugin.ldif | 12 ++++++++++++
 2 files changed, 13 insertions(+)
 create mode 100644 ldap/admin/src/scripts/50pbkdf2pwdstorageplugin.ldif

diff --git a/Makefile.am b/Makefile.am
index 8834a7819..055d480aa 100644
--- a/Makefile.am
+++ b/Makefile.am
@@ -949,6 +949,7 @@ update_DATA = ldap/admin/src/scripts/exampleupdate.pl \
 	ldap/admin/src/scripts/50refintprecedence.ldif \
 	ldap/admin/src/scripts/50retroclprecedence.ldif \
 	ldap/admin/src/scripts/50rootdnaccesscontrolplugin.ldif \
+	ldap/admin/src/scripts/50pbkdf2pwdstorageplugin.ldif \
 	ldap/admin/src/scripts/50contentsync.ldif \
 	ldap/admin/src/scripts/60upgradeschemafiles.pl \
 	ldap/admin/src/scripts/60upgradeconfigfiles.pl \
diff --git a/ldap/admin/src/scripts/50pbkdf2pwdstorageplugin.ldif b/ldap/admin/src/scripts/50pbkdf2pwdstorageplugin.ldif
new file mode 100644
index 000000000..462d5534a
--- /dev/null
+++ b/ldap/admin/src/scripts/50pbkdf2pwdstorageplugin.ldif
@@ -0,0 +1,12 @@
+dn: cn=PBKDF2_SHA256,cn=Password Storage Schemes,cn=plugins,cn=config
+objectclass: top
+objectclass: nsSlapdPlugin
+cn: PBKDF2_SHA256
+nsslapd-pluginpath: libpwdstorage-plugin
+nsslapd-plugininitfunc: pbkdf2_sha256_pwd_storage_scheme_init
+nsslapd-plugintype: pwdstoragescheme
+nsslapd-pluginenabled: on
+nsslapd-pluginDescription: DESC
+nsslapd-pluginVersion: PACKAGE_VERSION
+nsslapd-pluginVendor: VENDOR
+nsslapd-pluginid: ID
-- 
2.13.6

